c7n/resources/route53.py

```diff
@@ -267,24 +330,27 @@ def delete_records(self, client, hz):
             # Exempt the two zone associated mandatory records
             if rrset['Name'] == hz['Name'] and rrset['Type'] in ('NS', 'SOA'):
                 continue
+            rrsetdata = self.generate_rrset(rrset)
```

Comment:
it seems like since we now have a delete record set action, we could remove that implementation from here and just call the delete action on the record set, composition style to avoid the duplication.

Reply:
@kapilt 
Can you check and confirm, are below the expected change?

on @HostedZone.action_registry.register('delete')
class Delete(BaseAction):
```
 def process(self, hosted_zones):
        client = local_session(self.manager.session_factory).client('route53')
        rrsetdelete = ResourceRecordSetRemove(self.data, self.manager, self.log_dir)
        error = None
        for hz in hosted_zones:
            if self.data.get('force'):
                rrsetdelete.process("", hz)
```
and remove the method **def delete_records(self, client, hz):**

add equivalent entry on record set delete action like below 
```
  def process(self, recordsets, hz=None):
        client = local_session(self.manager.session_factory).client('route53')
        if hz is not None:
            paginator = client.get_paginator('list_resource_record_sets')
            paginator.PAGE_ITERATOR_CLS = RetryPageIterator
            recordsets = paginator.paginate(HostedZoneId=hz['Id']).build_full_result()

        try:
```

Reply:
In this composition style, we can avoid two delete implementation for record deletion.